packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/CopyFromRealmTests.kt

```diff
@@ -0,0 +1,755 @@
+            if (prop.type is SetPropertyType) {
+                val accessor: KMutableProperty1<BaseRealmObject, Any?> = fields[prop.name] as KMutableProperty1<BaseRealmObject, Any?>
+                val set: Set<Any?> = createPrimitiveSetData(prop, accessor)
+
+                if (prop.type.storageType == RealmStorageType.BINARY) {
+                    val copy = accessor.get(unmanagedCopy) as Set<ByteArray?>
+                    assertEquals(set.size, copy.size)
+                    copy.forEach { copiedValue: ByteArray? ->
+                        // Order is not guaranteed in the set when round-tripped through Core.
+                        // Also HashSets on JVM are rather annoying when it comes to byte arrays.
+                        // ByteArray equals/hashcode only considers the memory address, and not
+                        // the full content, so when copying byte arrays, the JVM does not consider
+                        // them equals. So for this test, use `any` instead of `contains`.
+                        if (copiedValue == null) {
+                            assertTrue(set.contains(copiedValue))
+                        } else {
+                            assertTrue(
+                                set.any {
+                                    (it as ByteArray).contentEquals(copiedValue)
+                                },
+                                "${prop.name} failed: $copiedValue"
+                            )
+                        }
+                    }
+                } else {
+                    val copiedSet = accessor.get(unmanagedCopy) as Set<Any?>
+                    assertEquals(set.size, copiedSet.size)
+                    copiedSet.forEach { copiedValue ->
+                        // Order is not guaranteed in the set when round-tripped through Core.
+                        assertTrue(set.contains(copiedValue), "${prop.name} failed: $copiedValue")
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    fun objectSet() {
+        val sample = Sample().apply {
+            objectSetField = (1..5).map { i ->
+                Sample().apply { stringField = i.toString() }
+            }.toRealmSet()
+        }
+
+        val insertedObj = realm.writeBlocking {
+            copyToRealm(sample)
+        }
+        val unmanagedObj: Sample = insertedObj.copyFromRealm()
+
+        // Close Realm to ensure data is decoupled from Realm
+        realm.close()
+
+        assertNotSame(insertedObj, unmanagedObj)
+        assertNotNull(unmanagedObj.objectSetField)
+        val copiedSet: RealmSet<Sample> = unmanagedObj.objectSetField
+        assertEquals(5, copiedSet.size)
+        copiedSet.forEach { copiedObject ->
+            assertEquals(
+                1,
+                sample.objectSetField.filter {
+                    copiedObject.stringField == it.stringField
+                }.size
+            )
+        }
```

Comment:
Again just remove with a predicate.

Reply:
Not sure what you mean, since this is copied objects, using `RealmObject.equals` is not going to work?

Reply:
Same as https://github.com/realm/realm-kotlin/pull/1092#discussion_r1022559921, but probably just safer just to leave as is.